media/js/wiki-samples.js

```diff
@@ -0,0 +1,90 @@
+(function(win, doc, $) {
+    if(!win.waffle || !win.waffle.flag_is_active('wiki_samples')) return;
+
+    var sites = ['jsfiddle', 'codepen'];
+    var frameLength = 'frame_'.length;
+
+    // using id to get sample code section since macros discard other attributes
+    $('.sample-code-frame').after(function() {
+        var section = $(this).attr('id').substring(frameLength);
+        return createSampleButtons(section, sites);
+    });
+    $('.sample-code-table').after(function(){
+        var section = $(this).find('iframe').attr('id').substring(frameLength);
+        return createSampleButtons(section, sites);
+    });
+
+    $('#wikiArticle').on('click', 'button.open-in-host', function(){
+        var $button = $(this);
+        var section = $button.attr('data-section');
+        var sampleCodeHost = $button.attr('data-host');
+        var sampleUrl = win.location.href.split('#')[0] + '?section=' + section + '&raw=1';
+
+        // track the click and sample code host
+        mdn.analytics.trackEvent({
+            category: 'Samples',
+            action: 'open-' + sampleCodeHost,
+            label: section
+        });
```

Comment:
I definitely don't have any disagreements with these choices, but also want to invite @groovecoder to share any thoughts. Custom event naming is an art... a confusing art. Like performance art or something.